### PR TITLE
Remove obsolete commands in CI (post-release fixes)

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Compatibility stable workflows are failing after release v0.37.0.

**Description of the Change:**
Remove temporary steps from the CI, such as downgrading Scipy to <1.14 and installing Kokkos v4.2 for `lightning-version == 'stable'`.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-68100]

### CODE REVIEW

File: .github/workflows/tests_lgpumpi_python.yml [hunk 1]
```diff
@@ -112,10 +112,6 @@ jobs:
           python -m pip install custatevec-cu${{ matrix.cuda_version_maj }} mpi4py openfermionpyscf
           PL_BACKEND=lightning_qubit python -m pip install . -vv
 
-      - name: Downgrade Scipy (remove after release)
-        if: inputs.lightning-version == 'stable'
-        run: pip install --force-reinstall "scipy<1.14"
-
```
Comment (AmintorDusko) [inline .github/workflows/tests_lgpumpi_python.yml:118]: Great! Thanks!

File: .github/workflows/tests_lkcpu_python.yml [hunk 2]
```diff
@@ -244,8 +244,6 @@ jobs:
 
       - name: Run PennyLane-Lightning unit tests
         run: |
-          # TODO: Remove installing pytest-xdist with release v0.36.0
-          python -m pip install pytest-xdist
```
Comment (AmintorDusko) [inline .github/workflows/tests_lkcpu_python.yml:248]: Oops?

**Review by AmintorDusko — APPROVED** [review]
Thank you. @vincentmr!

File: .github/workflows/tests_lgpumpi_python.yml (hunk 1 above)
Comment (vincentmr) [reply .github/workflows/tests_lgpumpi_python.yml:118]: Thanks for fixing this in the first place.

Comment (multiphaseCFD) [reply .github/workflows/tests_lgpumpi_python.yml:118]: Is this related to the changes in symbol names brought in scipy-1.14?

File: .github/workflows/tests_lkcpu_python.yml (hunk 2 above)
Comment (multiphaseCFD) [reply .github/workflows/tests_lkcpu_python.yml:248]: Did we run py tests in parallel with `xdist` before?

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @vincentmr !

**Pull request merged**